Add tensor short form option to vespa-visit and vespa-get

vekterli opened the pull request:
@bjorncs please review
@jobergum FYI

Specified with `--tensor-short-form`. No single-char option alias, as short form output will be the default on Vespa 9 and we're running out of usable option characters for these tools anyway.

This resolves #23470

vekterli committed:
Add tensor short form output option to vespa-visit tool

Specified with `--tensor-short-form`. No single-char option alias,
as short form output will be the default on Vespa 9 and we're running
out of usable option characters for this tool anyway.

vekterli committed:
Add tensor short form output option to vespa-get

bjorncs commented on vespaclient-java/src/main/java/com/yahoo/vespavisit/VdsVisit.java:
```diff
@@ -348,6 +348,13 @@ protected static Options createOptions() {
 
+        // TODO Vespa 9 replace with --tensor-long-form ?
+        options.addOption(Option.builder()
+                .longOpt("tensor-short-form")
```
Consider not using dash in parameter name (the other parameters don't)

vekterli replied:
I considered it, but I think writingoptionslikethisispointlesslyhardtoread so I went for the "be the change you want to see" approach. Save for entirely replacing the tool it's not given how we can ever transition away from a (IMHO) very suboptimal naming strategy if it cannot be done gradually.

But it might be less horrible if I rename it to something like `--shorttensors` instead, so I think I'll go with that.

vekterli replied:
Renamed

bjorncs commented on vespaclient-java/src/main/java/com/yahoo/vespaget/CommandLineOptions.java:
```diff
@@ -131,6 +132,13 @@ private static Options createOptions() {
 
+        // TODO Vespa 9: replace with --tensor-long-form ?
+        options.addOption(Option.builder()
+                .longOpt(TENSOR_SHORT_FORM_OPTION)
```
See comment for vespa-visit

vekterli committed:
Rename short tensor output CLI option

`tensor-short-form` -> `shorttensors` to be in line with existing
option formatting.
